docs/design-principles.md

```diff
@@ -0,0 +1,94 @@
+
+### Arch principles
+
+#### `r_rolename_` prefixed variables
+
+**Aim**: All variables used within a role are `r_rolename` and/or `l_` prefixed
```

Comment:
Do we denote what a ``role`` variable and a ``local`` variable is anywhere? If so link to it, if not it could be defined in this proposal.